libs/media-player/src/lib/audio-clip-player.tsx

```diff
@@ -0,0 +1,101 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import BugReportIcon from '@mui/icons-material/BugReport';
+import LinkOffIcon from '@mui/icons-material/LinkOff';
+import { Box, CircularProgress, IconButton, Tooltip } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { DefaultPlayButton } from './default-clip-player-button';
+
+interface IPlayButtonProps {
+    onButtonClick: () => void;
+}
+
+export interface IPlayButton {
+    (props: IPlayButtonProps): JSX.Element;
+}
+
+export interface AudioClipPlayerProps {
+    audioUrl: string;
+    UserDefinedPlayButton?: IPlayButton;
+}
+
+enum AudioState {
+    loading = 'loading',
+    error = 'error',
+    canPlay = 'canPlay',
+    undefined = 'undefined',
+}
+
+export function AudioClipPlayer({ audioUrl, UserDefinedPlayButton }: AudioClipPlayerProps) {
+    const [audioState, setAudioState] = useState<AudioState>(AudioState.loading);
+
+    useEffect(() => {
+        const onCanPlay = () => {
+            setAudioState(AudioState.canPlay);
+        };
+
+        const onError = () => {
+            setAudioState(AudioState.error);
+        };
+
+        /**
+         * This audio instance will never be played. We are using it to test whether
+         * the URL is valid.
+         */
+        const testAudio = new Audio(audioUrl);
+
+        testAudio.addEventListener('canplaythrough', onCanPlay);
+
+        testAudio.addEventListener('error', onError);
+
+        if (!isNullOrUndefined(audioUrl)) {
+            const testAudio = new Audio(audioUrl);
+
+            testAudio.addEventListener('error', onError);
+        } else {
+            setAudioState(AudioState.undefined);
+        }
+    }, [audioUrl]);
+
+    const handleMediaPlayerIconClick = () => {
+        console.log(`attempting to play audio`);
+
+        new Audio(audioUrl).play().catch((err) => console.log(`failed to play audio: ${err}`));
+    };
+
+    const PlayButton = isNullOrUndefined(UserDefinedPlayButton)
+        ? DefaultPlayButton
+        : UserDefinedPlayButton;
+
+    if (audioState === AudioState.loading) return <CircularProgress />;
+
+    if (audioState === AudioState.canPlay)
+        return (
+            <Tooltip title={audioUrl}>
+                <span>
+                    <PlayButton key={audioUrl} onButtonClick={handleMediaPlayerIconClick} />
+                </span>
+            </Tooltip>
+        );
+    if (audioState === AudioState.undefined)
+        return (
+            <Box>
+                <Tooltip title={typeof audioUrl}>
+                    <IconButton color="primary">
+                        <LinkOffIcon />
+                    </IconButton>
+                </Tooltip>
+                Audio url is {typeof audioUrl}.
```

Comment:
Just a thought for down the road and errors for the user:  in a case like this we might just want to have the disabled audio button and then either a tooltip with the error, or some less intrusive way of error reporting to the user that they can use when talking to our technical support team :)  Just to make things a little more slick.